libbeat/tests/integration/framework.go

```diff
@@ -118,6 +119,37 @@ func NewBeat(t *testing.T, beatName, binary string, args ...string) *BeatProc {
 		stdout:     stdoutFile,
 		stderr:     stderrFile,
 	}
+	t.Cleanup(func() {
+		if !t.Failed() {
+			return
+		}
+		stderr, err := os.ReadFile(filepath.Join(tempDir, "stderr"))
+		if err != nil {
+			t.Logf("error reading stderr: %s", err)
+		}
+		t.Logf("stderr was: %s", string(stderr))
+
+		stdout, err := os.ReadFile(filepath.Join(tempDir, "stdout"))
+		if err != nil {
+			t.Logf("error reading stdout: %s", err)
+		}
+		t.Logf("stdout was: %s", string(stdout))
+
+		glob := fmt.Sprintf("%s-*.ndjson", filepath.Join(tempDir, beatName))
+		files, err := filepath.Glob(glob)
+		for _, f := range files {
+			contents, err := os.ReadFile(f)
+			if err != nil {
+				t.Logf("error reading %s: %s", f, err)
+			}
+			maxlen := 2048
+			if len(contents) >= maxlen {
+				t.Logf("Last %d bytes of %s:\n%s", maxlen, f, string(contents[len(contents)-maxlen:]))
+			} else {
+				t.Logf("%s\n%s", string(contents), f)
+			}
```

Comment:
I'd rather stat the file, seek to `size-maxlen` and read to EOF. In case the files were very large it would impact the test time significantly.

Reply:
updated